### ISSUE TITLE
vagrant up hanging after "Setting the name of the VM"

### ISSUE THREAD

**Issue by LittleBigDev** [puphpet/puphpet#2262]
Hi all,
First fresh vagrant up seems to be hanging (more than 2 hours) after setting the name of the VM. No error message, no warning, nothing. When I try to ctrl + C, it's waiting for cleanup before exiting but..... hanging again and never exiting :(
Any help appreciated !!

Here's my config.yaml : 
```yaml
vagrantfile:
    target: local
    vm:
        provider:
            local:
                box: puphpet/debian75-x64
                box_url: puphpet/debian75-x64
                box_version: '0'
                chosen_virtualizer: virtualbox
                virtualizers:
                    virtualbox:
                        modifyvm:
                            natdnshostresolver1: false
                        showgui: 0
                    vmware:
                        numvcpus: 1
                    parallels:
                        use_linked_clone: 0
                        check_guest_tools: 0
                        update_guest_tools: 0
                machines:
                    vflm_i768be5z25ja:
                        id: dev.windows
                        network:
                            private_network: 192.168.56.101
                            forwarded_port:
                                vflmnfp_yigdmk0wx35e:
                                    host: '8185'
                                    guest: '22'
                        hostname: dev.puphpet
                        memory: '512'
                        cpus: '1'
        provision:
            puppet:
                manifests_path: puphpet/puppet/manifests
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
        synced_folder:
            vflsf_ss3qgb8ag0d7:
                source: 'c:/vagrant/webroot'
                target: /var/www
                sync_type: smb
                smb:
                    smb_host: ''
                    smb_username: ''
                    smb_password: ''
                    mount_options:
                        dir_mode: '0775'
                        file_mode: '0664'
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                        - .git/
                    auto: 'true'
                owner: www-data
                group: www-data
        usable_port_range:
            start: 10200
            stop: 10500
        post_up_message: ''
    ssh:
        host: 'false'
        port: 'false'
        private_key_path: 'false'
        username: vagrant
        guest_port: 'false'
        keep_alive: '1'
        forward_agent: 'false'
        forward_x11: 'false'
        shell: 'bash -l'
        insert_key: 'false'
    vagrant:
        host: detect
    proxy:
        http: ''
        https: ''
        ftp: ''
        no_proxy: ''
        enabled: ''
server:
    install: '1'
    packages:
        - vim
        - htop
users_groups:
    install: '1'
    groups: {  }
    users:
        - john.doe
locale:
    install: '1'
    settings:
        default_locale: fr_FR.UTF-8
        locales:
            - en_GB.UTF-8
            - en_US.UTF-8
        timezone: Europe/Paris
firewall:
    install: '1'
    rules: {  }
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
        proxy_connect_timeout: 600s
        proxy_send_timeout: 600s
        proxy_read_timeout: 600s
        names_hash_bucket_size: 128
    upstreams: {  }
    vhosts:
        nxv_m55grjf7a8vu:
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_q9zct90xnl7e:
                    location: /
                    autoindex: 'off'
                    internal: 'false'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                    proxy: ''
                    proxy_redirect: ''
                nxvl_7s0l0b215gid:
                    location: '~ \.php$'
                    autoindex: 'off'
                    internal: 'false'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
                    set:
                        - '$path_info $fastcgi_path_info'
                    proxy: ''
                    proxy_redirect: ''
    proxies: {  }
apache:
    install: '1'
    settings:
        version: 2.4
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - rewrite
    vhosts:
        av_wmbggzzcwgnw:
            servername: client.company.local
            serveraliases:
                - client.company.local
            docroot: /var/www/client-company
            port: '80'
            setenv:
                - 'APP_ENV dev'
            setenvif:
                - 'Authorization "(.*)" HTTP_AUTHORIZATION=$1'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
php:
    install: '1'
    settings:
        version: '56'
    modules:
        php:
            - cli
            - intl
            - curl
            - exif
            - gd2
            - gettext
            - gmp
            - imap
            - ldap
            - mbstring
            - mysql
            - mysqli
            - openssl
            - pdo_mysql
            - soap
            - sockets
            - xmlrpc
            - xsl
        pear: {  }
        pecl: {  }
    ini:
        display_errors: 'On'
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: Europe/Paris
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools: {  }
    composer: '1'
    composer_home: ''
xdebug:
    install: '1'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
blackfire:
    install: '0'
    settings:
        server_id: ''
        server_token: ''
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.21.1
drush:
    install: '0'
    version: 6.3.0
ruby:
    install: '1'
    versions: {  }
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '0'
    settings:
        version: '5'
    npm_packages: {  }
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mariadb:
    install: '0'
    settings:
        version: '10.1'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_ie889w4aap5w:
            name: dbuser
            password: '123'
    databases:
        mariadbnd_ozj42pw55c1d:
            name: dbname
            sql: ''
    grants:
        mariadbng_773xov48kl5m:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
mysql:
    install: '1'
    settings:
        version: '5.5'
        root_password: rootpassword
        override_options: {  }
    adminer: 0
    users:
        mysqlnu_lx0f6xmjdbwo:
            name: myusername
            password: mypassword
    databases:
        mysqlnd_0tzanbze5akl:
            name: myDatabase
            sql: ''
    grants:
        mysqlng_vrf6tj9skg2g:
            user: myusername
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.4'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        bind_ip: 127.0.0.1
        port: '27017'
    globals:
        version: 2.6.0
    databases: {  }
redis:
    install: '0'
    settings:
        port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailhog:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '8025'
        path: /usr/local/bin/mailhog
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '11300'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 2.1.0
        java_install: true
    instances:
        esi_zzp4f8mqaltk:
            name: es-01
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'
```

**Comment by jtreminio** [comment]
Your machine hostname and ID should match.

Otherwise, this isn't a puphpet issue but a vagrant one. Try running `vagrant up --debug` and it'll let you know where it's having problems!

**Comment by LittleBigDev** [comment]
Hi Juan and thx for the quick answer.
I tried `vagrant up --debug` but unfortunately, it still hangs and I'm a bit lost when reading the debug output...
Last actions before it hangs are (added separating blank lines for convenience) :
```
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderVirtualBox::Action::PrepareNFSValidIds:0x254cd60>
 INFO subprocess: Starting process: ["C:/Program Files/Oracle/VirtualBox/VBoxManage.exe", "list", "vms"]
 INFO subprocess: Command not in installer, restoring original environment...

 INFO warden: Calling IN action: #<VagrantPlugins::SyncedFolderNFS::ActionCleanup:0x430d2f8>
 INFO nfs: Host doesn't support pruning NFS. Skipping.

 INFO warden: Calling IN action: #<Vagrant::Action::Builtin::SyncedFolderCleanup:0x46ea2a8>
 INFO subprocess: Starting process: ["C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\/powershell.EXE", "-NoProfile", "-ExecutionPolicy", "Bypass", "$PSVersionTable.PSVersion.Major"]
 INFO subprocess: Command not in installer, restoring original environment...
```
Well... looks like a VirtualBox issue. I tried first command by myself : `VBoxManage list vms`
Result is : window pops up warning me that "VirtualBox Command Line Tool stopped to work" (english translation by myself, might be slightly different message). I can see the signature of this issue, I'll make some googlings before asking for more help here :)

**Comment by chrisatomix** [comment]
I was having the exact same issue (which is how I found this thread), but when I ran `vagrant up --debug` I noticed something about the synced folders. 

I'm running the latest VirtualBox and Vagrant on Windows 7 x64. This is the first time I've tried to build a PuPHPet VM using SMB synced folders, and I noticed that @LittleBigDev is also using SMB. So I changed the `config.yaml` from:
`sync_type: smb`
to:
`sync_type: default`
and immediately the VM started up.

I went back to the PuPHPet page to see if there was something I missed, and I noticed there was a "[More information](https://www.vagrantup.com/docs/synced-folders/smb.html)" link when selecting the synced folder type. It said I needed PowerShell 3 or later, so I checked my version and it said I was still running PowerShell 2. I downloaded and installed PowerShell 4 from Microsoft, rebooted my PC, and tried `vagrant up` again. Problem fixed!

To check your PowerShell version:
# Open PowerShell
# Run $PSVersionTable.PSVersion
# If it says "1" or "2" in the "Major" column, then you need to update.

**Comment by LittleBigDev** [comment]
Not to overload this issue, I didn't tell about the PoserShell version. But I also upgraded my version, with same result : other issues. So I came back to default sync type. I also had to activate virtualization in my laptop BIOS. I still have some errors, but vagrant up finshes and I can ssh connect to it (if you are interrested, my other issues are listed here : https://github.com/puphpet/puphpet/issues/2265 )
